### PR TITLE
Add support for generating StaticType and Value trait impls for flags…

### PR DESCRIPTION
**Pull request by sdroege**
… and enums

### CODE REVIEW

File: src/codegen/flags.rs [hunk 1]
```diff
@@ -122,5 +151,43 @@ impl FromGlib<ffi::{ffi_name}> for {name} {{
...
+}}
+",
+            name = flags.name,
+            ffi_name = flags.c_type,
+            get_type = get_type
+        ));
+
+        try!(writeln!(
+            w,
+            ""));
```
Comment (EPashkin) [inline src/codegen/flags.rs]: This cause extra line.

File: src/codegen/flags.rs [hunk 2]
```diff
@@ -14,21 +14,50 @@ use traits::*;
...
+            if has_get_type {
+                try!(writeln!(w, "use glib::Type;"));
+                try!(writeln!(w, "use glib::StaticType;"));
+                try!(writeln!(w, "use glib::value::{{Value, SetValue, FromValue, FromValueOptional}};"));
+                try!(writeln!(w, "use glib_ffi;"));
+                try!(writeln!(w, "use gobject_ffi;"));
+            }
             try!(writeln!(w, "use glib::translate::*;"));
         }
+        try!(writeln!(w, "use std;"));
```
Comment (EPashkin) [inline src/codegen/flags.rs]: This seems uneeded

File: src/codegen/enums.rs [hunk 3]
```diff
@@ -245,6 +273,44 @@ impl FromGlib<ffi::{ffi_name}> for {name} {{
...
+}}
+",
+            name = enum_.name,
+            ffi_name = enum_.c_type,
+            get_type = get_type
+        ));
+
+        try!(writeln!(
+            w,
+            ""));
```
Comment (EPashkin) [inline src/codegen/enums.rs]: Also extra line

Comment (sdroege) [inline src/codegen/enums.rs]: Otherwise there is no extra line between this and the next block

File: src/codegen/flags.rs (hunk 1 above)
Comment (sdroege) [inline src/codegen/flags.rs]: Otherwise there is no extra line between this and the next block

File: src/codegen/enums.rs (hunk 3 above)
Comment (EPashkin) [inline src/codegen/enums.rs]: You you prefer 2 lines between enums?

**Comment by EPashkin** [comment]
Results many unused `glib_ffi` in pango. Maybe something wrong with conditions

File: src/codegen/enums.rs (hunk 3 above)
Comment (sdroege) [inline src/codegen/enums.rs]: Without this, I had no line between enums :) I'll check it again

**Comment by sdroege** [comment]
I'll wait with that until the other PRs are merged, the unused glib_ffi are probably related to that

**Comment by sdroege** [comment]
The unused warnings should be gone now, thanks!

**Comment by EPashkin** [comment]
Value impls for enums and flags don't have #cfg (see GdkGLError and GdkAxisFlags).
And still present extra lines.

**Comment by sdroege** [comment]
> Value impls for enums and flags don't have #cfg (see GdkGLError and GdkAxisFlags).

Fixed

> And still present extra lines.

Please run the code generator and look at the output now. Do you want this one empty line between impls to disappear?

**Comment by sdroege** [comment]
>  Run everything through rustfmt 0.1.9-nightly

And a small other commit for cleanup, seems a) things have improved in rustfmt and b) we didn't run it since a bit after changes.

File: src/analysis/out_parameters.rs [hunk 4]
```diff
@@ -98,8 +98,7 @@ pub fn analyze_imports(env: &Env, func: &Function, imports: &mut Imports) {
     for par in &func.parameters {
         if par.direction == ParameterDirection::Out {
             match *env.library.type_(par.typ) {
-                Type::Bitfield(..) |
-                Type::Enumeration(..) => imports.add("std::mem", func.version),
+                Type::Bitfield(..) | Type::Enumeration(..) => imports.add("std::mem", func.version),
```
Comment (EPashkin) [inline src/analysis/out_parameters.rs]: This is why I don't sometimes don't like rustfmt 😭

**Comment by EPashkin** [comment]
Seems Appveyor have bad connection to cargo (3 restart wont helps).
Let ignore it and merge.

File: src/analysis/out_parameters.rs (hunk 4 above)
Comment (GuillaumeGomez) [inline src/analysis/out_parameters.rs]: No one does.

**Pull request merged**